### PR TITLE
[feat] Query Active & Archived Sessions

### PR DESCRIPTION
**Pull request by dylantohenyu**
### 🦋 Description
- Created a unstyled query to retrieve "Sessions" data.
- It is split into "Archived" and "Active" Sessions, with each session displaying a respective list of their participants. 

### 🦧 Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior 

## 🐸 How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- Test for whether a specific session data comes up.
- Test to see if participants arise. 

## 🐙 Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- Check with VPT if we truly need to do a api/sessions/[Sessionid]/ or are we chopped. I am sorry Alex, we should ahve taken security vulnerabilities. 

### Online sources
PM me. 

### 🐁 Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @eshabansiya

### CODE REVIEW

File: api/supabase/queries/sessions.ts [hunk 1]
```diff
@@ -110,8 +110,9 @@ export type SessionParticipant = {
 // x.profile?.first_name and x.profile?.first_name
 // returns SessionParticipant[]
 export async function sessionParticipants(
-  session_id: UUID,
-): Promise<SessionParticipant[]> {
+  session_id: string, //dt: @esha, i had to change this to a string for my api,
+  // but if we need to change it back to UUID, i need to ask how to change the parameter type in the api route as well
```
Comment (eshabansiya) [inline api/supabase/queries/sessions.ts]: There is a new UUID that is formated as a string, so we can prob replace this back once you pull from main

File: app/api/sessions/[sessionID]/participants/route.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    return NextResponse.json(participants);
+  } catch (error) {
+    console.error("Failed to fetch participants", error);
+    return NextResponse.json(
+      { error: "Failed to fetch participants" },
+      { status: 500 }
+    );
+  }
+}
```
Comment (eshabansiya) [inline app/api/sessions/[sessionID]/participants/route.ts]: hey dylan! we dont need this file bc now we have the server queries set up so just 
const participants = await sessionParticipants(sessionID);

in the actual page!

File: app/participants/past-exercises/page.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+export default function PastSessionsPage() {
+  const [sessions, setSessions] = useState<Session[]>([]);
+  const [participantsBySession, setParticipantsBySession] = useState<
+    Record<string, ParticipantSession[]>
+  >({});
+  const [loading, setLoading] = useState(true);
+  // const [DEBUG_SESSION_ID] = "0385b601-78cf-4997-9a2d-9fb137637203";
+
+  useEffect(() => {
+    async function fetchSessions() {
```
Comment (eshabansiya) [inline app/participants/past-exercises/page.tsx]: update these like before to just get the data directly in this page, and also add a console.log for now for testing

File: app/participants/past-exercises/page.tsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                    <div>
+                      : {p.profile.first_name} {p.profile.last_name}
+                    </div>
+                  </li>
+                ))}
+              </ul>
+            </li>
+          ))}
+        </ul>
+      )}
```
Comment (eshabansiya) [inline app/participants/past-exercises/page.tsx]: we dont need all these html tags, you can just do a normal list cause this will be styled later

**Review by eshabansiya — CHANGES_REQUESTED** [review]
hi!! thank you for getting this in and all the randomness
i made some comments 
please pull from main first, cause it has all the updated stuff, it may have some conflicts cause all the queries are moved into actions
we dont need to api route for this cause now we have a server side, and if you can, add a screenshot of it showing some session, i wasnt really able to test it 
also the route to this page doesnt work rn so please update the route - its cause somewhere its going to particiapant/past-excercises not participantS with an s
